core/src/main/java/org/infinispan/configuration/parsing/Parser.java

```diff
@@ -2755,6 +2755,11 @@ private void parseIndexing(XMLExtendedStreamReader reader, ConfigurationBuilderH
          builder.indexing().enable();
       }
 
+      if (!builder.indexing().enabled()) {
+         // discard any eventually inherited indexing config if indexing is not enabled
+         builder.indexing().reset();
+      }
```

Comment:
This should be done in the builder itself, in ```IndexingConfigurationBuilder#create()```. The issue of doing here is it covers only XML parsing.

Reply:
indeed. but ```create()``` is too late because we cannot distinguish the inherited vs own. and we also don;t even have a ref to base config to try to figure out.

Reply:
what about resetting everything in the builder whenever the index is disabled?

Reply:
I've pushed a new WIP commit to try that.

This thing gets disgustingly tangled :octopus: